### PR TITLE
Custom certificate validator for OCES3 & comments

### PR DESCRIPTION
**Pull request by gpikmddk**
Based on the research for IdP Proxy I added similar handling to Test Application (SAML).
There is also plenty of comments on when to use a specific mode

### CODE REVIEW

File: KMD.Identity.TestApplications.SAML.MVCCore/appsettings.json [hunk 1]
```diff
@@ -7,15 +7,35 @@
     }
   },
   "Saml2": {
-    "IdPMetadata": "https://identity.kmd.dk/federationmetadata/2007-06/federationmetadata.xml",
```
Comment (JLJKMD) [inline KMD.Identity.TestApplications.SAML.MVCCore/appsettings.json:10]: Maybe we still want to keep this just out commented and with a note about when to use which IdPMetadata?

Comment (JLJKMD) [inline KMD.Identity.TestApplications.SAML.MVCCore/appsettings.json:10]: Should we maybe keep this and out comment it and leave a note about when to use which metadata?

Comment (gpikmddk) [reply KMD.Identity.TestApplications.SAML.MVCCore/appsettings.json:10]: ahh, good catch, I forgot to revert it :)

Comment (gpikmddk) [reply KMD.Identity.TestApplications.SAML.MVCCore/appsettings.json]: reverted to the previous

**Review by JLJKMD — APPROVED** [review]
I think the comments are nice.

File: KMD.Identity.TestApplications.SAML.MVCCore/Infrastructure/Saml/CustomSamlCertificateValidator.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+﻿using ITfoxtec.Identity.Saml2.Util;
+using Microsoft.IdentityModel.Tokens;
+using System.Security.Cryptography.X509Certificates;
+using System;
+using System.Linq;
+
+namespace KMD.Identity.TestApplications.SAML.MVCCore.Infrastructure.Saml
+{
+    /// <summary>
+    /// This is an extension to ItFoxtec certificate validator, which will fail when building chain in specific situations:
```
Comment (mci-kmd) [inline KMD.Identity.TestApplications.SAML.MVCCore/Infrastructure/Saml/CustomSamlCertificateValidator.cs:10]: This line confused me. At first I read it as this class failing in certain conditions, but as a read the rest of the PR I came the the conclusion that "which" refers to the base class. I think this is caused by the comma which makes the sentence incorrect. However, I suggest rephrasing this to something like the following for extra clarity:

"This is an extension to the ItFoxtec certificate validator which allows you to soften the validation requirements for certificates. The following scenarios will cause certificate validation to fail in the base class but can be ignored in this class using the validation flag:"

**Pull request merged**